tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -2232,31 +2233,36 @@ public:
    ///
    ColumnNames_t GetColumnNames()
    {
-      ColumnNames_t allColumns;
+      // there could be duplicates between Redefined columns and columns in the data source
+      std::unordered_set<std::string> allColumns;
 
       auto addIfNotInternal = [&allColumns](std::string_view colName) {
          if (!RDFInternal::IsInternalColumn(colName))
-            allColumns.emplace_back(colName);
+            allColumns.emplace(colName);
       };
 
-      auto columnNames = fColRegister.GetNames();
+      auto definedColumns = fColRegister.GetNames();
 
-      std::for_each(columnNames.begin(), columnNames.end(), addIfNotInternal);
+      std::for_each(definedColumns.begin(), definedColumns.end(), addIfNotInternal);
```

Comment:
Another cplusplusic way of writing this is
```suggestion
      std::unordered_set<std::string> allColumns(definedColumns.begin(), definedColumns.end());
```
and then insert everything else. Once done, do a pass of `remove_if(allColumns.begin(), allColumns.end(), isInternalColumn());`

Reply:
I just learned that the erase-remove idiom (or even just `std::remove_if`) does not work with `unordered_set` & friends: https://stackoverflow.com/questions/3792600/erase-remove-idiom-with-stdset-failing-with-constness-related-error